test/tstDetailsTypeRequirements.cpp

```diff
@@ -0,0 +1,65 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace Test
+{
+// NOTE only supporting Point and Box
+using FakePrimitive = ArborX::Point;
```

Comment:
Could you change this to something that is not ArborX? I think typedefing to `Point` or `Box` may hide some type requirements. I do see you note, but I don't fully understand its meaning hear (I assume that it is the return type of `AccessTraits`). I would rather do a special type with specializing `AccessTraits` for it.

Reply:
Supporting anything else is more involved.  One of the issues is at the moment is the `Box::operator+=(Primitive)` requirement.  Also I like that we force the user to provide `Point` or `Box` because in a problem like the ray tracing example in #461 one could be surprised if `Sphere` was given but ArborX in fact computed intersection with the `Box` that contains it.

Reply:
OK. After talking to @dalg24, what it really means hear is not the primitive as in user geometry sense, but rather a primitive as something that `BVH` can be constructed on, i.e. `Point` or `Box`. Used in this sense, it's not really fake. This test also do not strive to answer the question: if we wanted to build a hierarchy on a new type of primitive, say `kDOP`, what are the requirements for that type. This PR only addresses the requirements on bounding volume types (template argument in `BasicBoundingVolumeHierarchy`) and predicates.

Thus, with this understanding, I mark this as resolved.